src/pages/index.js

```diff
@@ -43,73 +106,189 @@ const initialCards = [
+  button.disabled = isLoading;
+  button.classList.toggle("modal__submit-btn_loading", isLoading);
+  button.textContent = isLoading ? loadingText : defaultText;
+}
 
-  cardLikedBtn.addEventListener("click", () => {
-    cardLikedBtn.classList.toggle("card__like-btn_liked");
-  });
+function getCardElement(cardData) {
+  const cardTemplate = document.querySelector("#card-template").content;
+  const cardElement = cardTemplate.querySelector(".card").cloneNode(true);
+
+  // Add a unique ID to the card
+  cardElement.id = `card-${cardData._id}`;
+
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardTitle = cardElement.querySelector(".card__title");
+  const cardLikeButton = cardElement.querySelector(".card__like-btn");
+  const deleteButton = cardElement.querySelector(".card__delete-btn");
+
+  cardImage.src = cardData.link;
+  cardImage.alt = cardData.name;
+  cardTitle.textContent = cardData.name;
 
   deleteButton.addEventListener("click", () => {
-    cardElement.remove();
+    handleDeleteCard(cardElement, cardData);
   });
 
-  cardImageElement.addEventListener("click", () => {
-    openModal(previewModal);
+  // Set initial like state
+  if (cardData.isLiked) {
+    cardLikeButton.classList.add("card__like-btn_liked");
+  }
 
-    previewModalImageElement.src = data.link;
-    previewModalImageElement.alt = data.name;
-    previewModalCaptionElement.textContent = data.name;
+  // Open the card preview
+  cardImage.addEventListener("click", () => {
+    previewModalImageElement.src = cardData.link;
+    previewModalImageElement.alt = cardData.name;
+    previewModalCaptionElement.textContent = cardData.name;
+    openModal(previewModal);
   });
 
+  // Add click event listener for like button
+  cardLikeButton.addEventListener("click", () => {
+    handleLikeClick(cardData._id, cardLikeButton);
+  });
   return cardElement;
 }
 
+// Open Avatar Modal
+profileAvatarBtn.addEventListener("click", () => {
+  resetValidation(avatarForm, settings);
+  openModal(avatarModal);
+});
+
+// Close Avatar Modal
+avatarModalCloseBtn.addEventListener("click", () => closeModal(avatarModal));
+
+// Avatar Form Submission
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
 function openModal(modal) {
   modal.classList.add("modal_opened");
   document.addEventListener("keydown", handleEscapeKey);
 }
+// Handle avatar update submission
+// Avatar submission handler
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = avatarForm.querySelector(".modal__submit-btn");
+  const avatarImg = document.querySelector(".profile__avatar");
+
+  setSubmitButtonState(submitButton, true, "Saving...", "Save");
+
+  api
+    .updateAvatar({ avatar: avatarUrlInput.value })
+    .then((userData) => {
+      avatarImg.src = `${userData.avatar}?t=${Date.now()}`;
+      avatarForm.reset();
+      closeModal(avatarModal);
+      resetValidation(avatarForm, settings);
```

Comment:
`resetValidation` is not needed after **submission** because there can't be any validation errors if the submission is successful